src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java

```diff
@@ -841,12 +842,12 @@ public boolean isShadow() {
 	public <E extends CtShadowable> E setShadow(boolean isShadow) {
 		getFactory().getEnvironment().getModelChangeListener().onObjectUpdate(this, IS_SHADOW, isShadow, this.isShadow);
 		this.isShadow = isShadow;
-		return (E) this;
+		return Unchecked.castToReturnType(this);
 	}
 
 	@Override
 	public CtTypeReference<T> clone() {
-		return (CtTypeReference<T>) super.clone();
+		return Unchecked.castToReturnType(super.clone());
```

Comment:
@SirYwell This also seems like an appropriate place to use this, wdyt?

Reply:
Yep, that makes sense aswell.